Shared/Supporting Files/Views/DownloadOfflineResourcesView.swift

```diff
@@ -0,0 +1,172 @@
+    
+    /// The current state of the "download all on-demand resources" request.
+    @State private var downloadAllRequestState: OnDemandResource.RequestState?
+    
+    /// A Boolean value indicating whether confirm cancel alert is showing.
+    @State private var confirmCancelAlertIsShowing = false
+    
+    /// A Boolean value indicating whether all of the `onDemandResources` have successfully downloaded.
+    private var allResourcesAreDownloaded: Bool {
+        onDemandResources.allSatisfy { $0.requestState == .downloaded }
+    }
+    
+    var body: some View {
+        NavigationStack {
+            Form {
+                Section {
+                    Button {
+                        downloadAllRequestState = .inProgress
+                    } label: {
+                        Label {
+                            Text("Download All")
+                        } icon: {
+                            RequestStateView(state: downloadAllRequestState)
+                        }
+                        .frame(maxWidth: .infinity)
+                    }
+                    .disabled(downloadAllRequestState != .notStarted || allResourcesAreDownloaded)
+                    .task(id: downloadAllRequestState) {
+                        if downloadAllRequestState == nil {
+                            await setUpResources()
+                        } else if downloadAllRequestState == .inProgress {
+                            await downloadAll()
+                        }
+                    }
+                    .onChange(of: allResourcesAreDownloaded) {
+                        guard allResourcesAreDownloaded else { return }
+                        downloadAllRequestState = .downloaded
+                    }
+                }
+                
+                ForEach(onDemandResources, id: \.sampleName) { resource in
+                    DownloadOnDemandResourceView(resource: resource)
+                }
+            }
+            .navigationTitle("Download Offline Resources")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") {
+                        if onDemandResources.allSatisfy({ $0.requestState != .inProgress }) {
+                            dismiss()
+                        } else {
+                            confirmCancelAlertIsShowing = true
+                        }
+                    }
+                    .alert("Cancel Downloads?", isPresented: $confirmCancelAlertIsShowing) {
+                        Button("Resume", role: .cancel, action: {})
+                        
+                        Button("Confirm") {
+                            for resource in onDemandResources where resource.requestState == .inProgress {
+                                resource.cancel()
+                            }
+                            dismiss()
+                        }
+                    }
```

Comment:
I'm not sure this alert is entirely necessary. But if it is, here are some suggestions related to the HIG:

- The title shouldn't use title case since it is a question ("Cancel downloads?")
- The button titles should be more related to the question in the title: "Don't Cancel" and "Cancel" would be good